models/public/midasnet/model.yml

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) 2019 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+description: >-
+  MidasNet is a model for monocular depth estimation trained by mixing several datasets;
+  as described in the following paper: 
+  "Towards Robust Monocular Depth Estimation: Mixing Datasets for Zero-Shot Cross-Dataset Transfer"
+  <https://arxiv.org/pdf/1907.01341>
+
+  The model input is a blob that consists of a single image of "1x3x384x384" in RGB order.
+
+  The model output is an inverse depth map that is defined up to an unknown scale factor.
+task_type: mono_depth
+files:
+  - name: midasnet.py
+    size: 6464
+    sha256: a3680912f1e6839b5bc4d4684e78b2a8f5f669125ee86aff81b53dbe8d635a12
+    source:
+      $type: google_drive
+      id: 1fL_UDsZ8pFUN3QnXqMF4BReVZmfmrKBj
```

Comment:
Is this code different in any way from what's in the repository?

Reply:
The code is different since our base model can not be converted to dldt via onnx. Some changes were required to the original code; such as fixed image size, changes in upsampling.

Reply:
I took a look at the changes you made to the original code. Other than concatenation, there were four changes:

1. Replacing `torch.hub.load` with a torchvision model.
2. Replacing scale factors for interpolation with sizes (and specifying explicit sizes everywhere).
3. Replacing `align_corners=True` with `align_corners=False` in `FeatureFusionBlock`.
4. Replacing `use_pretrained = False if path else True` with just `use_pretrained = False`.

Now, I've managed to convert the model by taking the original code and applying changes 1, 3 and 4. Change 2 doesn't actually seem necessary. Maybe it was only needed for older PyTorches and/or Model Optimizers?

Changes 3 and 4 are simple enough that you can do them through Model Downloader itself (just search for `regex_replace` in other model configs to see how this can be done). Change 1 looks like you could probably just apply it in the original repository, but if you don't want to, then you can also do it with `regex_replace`.

All this means that you can change the model config to get the sources from the original repository, and I would prefer if you did so. This would be better for a few reasons:

1. It would make it clear where the code came from to someone looking at the config.
2. It would explicitly enumerate the changes that have to be made to the code in order to convert it to the IR format.
3. Change 4 might become unnecessary in the future if Model Optimizer supports ONNX opset 11, in which case it would be prudent to remove it. This will be a lot easier to do if it's applied via a post-processing step during downloading.

Reply:
I would klike to debug this, but the download of the weight from the gdrive is still not working; see https://github.com/opencv/open_model_zoo/pull/843#issue-373737647

Reply:
I tested the downloading / conversion with a local webserver. Works fine. I commited the changes. Remainign issue: downloading from gdrive fails.

Reply:
I actually can't download it myself either anymore without logging into Google. Google just says that the file has been downloaded too many times and is now unavailable. Not sure we can do anything about this but wait...

(That said, for debugging the downloading of the model _code_ you could probably just comment out the weights file in the config and place the weights file into the model directory manually.)

Reply:
Downloading via chrome works fine for me. 
However, downloading via the downloader tool not.

Maybe at makes sense to test the download tool w/ and w/o being logged-in to google.

Reply:
> Downloading via chrome works fine for me.

Even in an incognito window?

> It might make sense to test the download tool w/ and w/o being logged-in to google.

The tool can't itself log in, and your browser login shouldn't make a difference.

Reply:
That is correct. Then, I do not understand why the download via the tool is/was not working for you, but for me.

Reply:
It's a good question, but now it works for neither of us, so it's kinda of moot. :-) For now I'm just going to wait 24 hours as the Google Drive message suggests and see if the situation improves.

(I'm also going to resolve this conversation, since my original comment is no longer relevant.)